File: mobile_app/lib/data/content_repository.dart
import 'api_client.dart';
import 'models/content_models.dart';
import 'models/story_models.dart';
import 'models/avatar_models.dart';

class ContentRepository {
  final ApiClient _apiClient;

  ContentRepository({ApiClient? apiClient}) : _apiClient = apiClient ?? ApiClient();

  Future<List<AvatarItem>> fetchAvatars() async {
    final data = await _apiClient.getJsonList('/content/avatars/all');
    return data.map((e) => AvatarItem.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<ContentListResponse> fetchContentPaged({
    required String type,
    required String language,
    int page = 1,
    int pageSize = 20,
  }) async {
    final data = await _apiClient.getJson(
      '/content/paged',
      query: {
        'type': type,
        'language': language,
        'page': page.toString(),
        'pageSize': pageSize.toString(),
      },
    );
    return ContentListResponse.fromJson(data);
  }

  Future<ContentDetail> fetchContentById(String id) async {
    final data = await _apiClient.getJson('/content/$id');
    return ContentDetail.fromJson(data);
  }

  Future<StoryListResponse> fetchStoriesPaged({
    required String language,
    int page = 1,
    int pageSize = 20,
  }) async {
    final data = await _apiClient.getJson(
      '/content/stories',
      query: {
        'language': language,
        'page': page.toString(),
        'pageSize': pageSize.toString(),
      },
    );
    return StoryListResponse.fromJson(data);
  }

  Future<StoryPageResponse> fetchStoryPage({
    required String storyId,
    required int pageNumber,
  }) async {
    final data = await _apiClient.getJson('/content/stories/$storyId/pages/$pageNumber');
    return StoryPageResponse.fromJson(data);
  }
}

File: mobile_app/src/data/contentRepository.ts
import { apiClient } from './apiClient';
import {
  parseAvatar,
  parseContentDetail,
  parseContentList,
  parseStoryList,
  parseStoryPageResponse,
  type AvatarItem,
} from './models';

export const contentRepository = {
  async fetchAvatars(): Promise<AvatarItem[]> {
    const data = await apiClient.getJsonList('/content/avatars/all');
    return data.map((item) => parseAvatar(item as Record<string, unknown>));
  },

  async fetchContentPaged(input: {
    type: string;
    language: string;
    page?: number;
    pageSize?: number;
  }) {
    const data = await apiClient.getJson('/content/paged', {
      type: input.type,
      language: input.language,
      page: String(input.page ?? 1),
      pageSize: String(input.pageSize ?? 20),
    });
    return parseContentList(data);
  },

  async fetchContentById(id: string) {
    const data = await apiClient.getJson(`/content/${id}`);
    return parseContentDetail(data);
  },

  async fetchStoriesPaged(input: {
    language: string;
    page?: number;
    pageSize?: number;
  }) {
    const data = await apiClient.getJson('/content/stories', {
      language: input.language,
      page: String(input.page ?? 1),
      pageSize: String(input.pageSize ?? 20),
    });
    return parseStoryList(data);
  },

  async fetchStoryPage(storyId: string, pageNumber: number) {
    const data = await apiClient.getJson(
      `/content/stories/${storyId}/pages/${pageNumber}`,
    );
    return parseStoryPageResponse(data);
  },
};
